app/stac_api/serializers/utils.py

```diff
@@ -339,3 +343,21 @@ def to_representation(self, value):
 
     def to_internal_value(self, data):
         return data
+
+
+class IsoDurationField(serializers.Field):
+    '''Handles duration in the ISO 8601 format like "P3DT6H"'''
+
+    def to_internal_value(self, data):
+        '''Convert from ISO 8601 (e.g. "P3DT1H") to Python's timedelta'''
+        internal = parse_duration(data)
+        if internal is None:
+            raise serializers.ValidationError(
+                code="payload",
+                detail={'href': _("Duration doesn't match ISO 8601 format")}
```

Comment:
I guess is should be `'duration'` not `'href'`?

Reply:
Yes, will fix. Lack of understanding from my part.

While the actual field name would be better, we agreed that only having the error message is good enough. Having `duration` would only be confusing because there is no such field.